### ISSUE TITLE
New decrypt message UI

### ISSUE THREAD

**Issue by logvik** [MetaMask/Design#149]
Screenshot for https://github.com/MetaMask/metamask-extension/issues/1190#issuecomment-543091839
![decrypt_request](https://user-images.githubusercontent.com/2010097/66999589-fbd5a780-f0de-11e9-99b6-2d79b3ff2c83.jpg)

**Comment by danfinlay** [comment]
Some experience questions:

- If the user approves this, the decrypted result is passed back to the requesting site? If so, we should indicate at the top "The domain X wants to decrypt a message intended for account Y"

I'm not sure showing the encrypted blob is useful! It's nearly useless by definition! Maybe parsing out that metadata a bit, to name the type of encryption, and the length of the message, but otherwise it just makes for a more confusing message.

Once we deploy the [Login permissions branch](https://github.com/MetaMask/metamask-extension/pulls) we could extend this to a blanket permission, like "Allow domain X to decrypt messages intended for account Y".

**Comment by danfinlay** [comment]
Rather than showing the encrypted blob, we could even have a "peek at decrypted value" button, so the user could view the decrypted message right there.

The button is more specifically about giving that data back to the site, so I'm trying to think of a good button name for that. Maybe "Approve" is fine if the initial prompt is worded well enough.

**Comment by logvik** [comment]
> If the user approves this, the decrypted result is passed back to the requesting site? If so, we should indicate at the top "The domain X wants to decrypt a message intended for account Y"

Yes, the same as for the signing message process. But here we can add this "Note".

> so the user could view the decrypted message right there.

Agreed, then need to add an additional step -> decrypt the message -> share with the website.
Maybe for some cases will be preferably taking a look at the message right in the extension and don't share the source text.

**Comment by logvik** [comment]
For `encryption_public_key` method we can use the same window with changes:

**Title**: Encryption public key request

**Content**: This action will allow using the generated public key for encryption messages in Dapp. This key will be generated via your private key, but can't be used on Ethereum blockchain.

**Buttons**: Cancel / Generate

**Comment by logvik** [comment]
@danfinlay sorry for disturbing, but maybe you will have some a little time to review updates in PR in all subprojects?

**Comment by danfinlay** [comment]
Sorry I'm so behind on emails. There are updates? Want to list out the links here for my convenience so I can knock them out quickly?

**Comment by logvik** [comment]
all updates are attached to https://github.com/MetaMask/metamask-extension/issues/1190

**Comment by lazaridiscom** [comment]
@logvik, please allow me to jump in:

> Want to list out the links here for my convenience 

Trivial

* https://github.com/MetaMask/eth-simple-keyring/pull/32 => Review, Merge
* https://github.com/MetaMask/KeyringController/pull/41 => Review, Merge

More effort

* https://github.com/MetaMask/metamask-extension/pull/7247 => Review, e2e-test, Merge

**Comment by lazaridiscom** [comment]
related: #154

**Comment by danfinlay** [comment]
I've merged the non-design related PRs. Just waiting on design & UI finalization to merge #7247.
